index.bs

```diff
@@ -485,6 +501,159 @@ spec:csp3; type:grammar; text:base64-value
+     |integrity policy|'s <a>sources</a>.
+  1. If |dictionary|["`blocked-destinations`"] <a for=list>exists</a>:
+    1. If its value <a for=list>contains</a> "`script`",
+       <a for=list>append</a> "`script`" to |integrity policy|'s <a>blocked destinations</a>.
+  1. If |dictionary|["`endpoints`"] <a for=map>exists</a>:
+    1. Set |integrity policy|'s <a>endpoints</a> to |dictionary|['endpoints'].
+  1. Return |integrity policy|.
+  
+  ### Parse Integrity-Policy headers ### {#parse-integrity-policy-headers-section}
+  To <dfn>parse Integrity-Policy headers</dfn>, given a <a for=/>Response</a> |response|
+  and a <a for=/>policy container</a> |container|, do the following:
+
+  1. Let |headers| be |response|'s <a for=response>header list</a>.
+  1. If |headers| <a for="header list">contains</a> a <a>header</a>
+     whose <a>header name</a> is "`Integrity-Policy`",
+     set |container|'s <a>integrity policy</a> be the result of running
+     <a>processing an integrity policy</a> with the corresponding <a>header value</a>.
+  1. If |headers| <a for="header list">contains</a> a <a>header</a>
+     whose <a>header name</a> is "`Integrity-Policy-Report-Only`",
+     set |container|'s <a>report only integrity policy</a> be the result of running
+     <a>processing an integrity policy</a> with the corresponding <a>header value</a>.
+  
+  ### Should request be blocked by Integrity Policy ### {#should-request-be-blocked-by-integrity-policy-section}
+  To determine <dfn>should request be blocked by integrity policy</dfn>, given a <a for=/>request</a> |request|,
+  do the following:
+
+  1. Let |policy container| be |request|'s <a for=request>policy container</a>.
+  1. If |request|'s <a>integrity metadata</a> is not the empty string
+     and |request|'s <a for=request>mode</a> is either "cors" or "same-origin",
+     return "Allowed".
+  1. Let |policy| be |policy container|'s <a>integrity policy</a>.
+  1. Let |report policy| be |policy container|'s <a>report only integrity policy</a>.
+  1. If both |policy| and |report policy| are empty <a>integrity policy struct</a>s, return "Allowed".
+  1. Let |global| be |request|'s <a for=request>client</a>'s <a for="environment settings object">global object</a>.
+  1. If |global| is not a {{Window}} nor a {{WorkerGlobalScope}}, return "Allowed".
+  1. Let |block| be a boolean, initially false.
+  1. Let |report block| be a boolean, initially false.
+  1. If |policy|'s <a>sources</a> <a for=list>contains</a> "`inline`"
+     and |policy|'s <a>blocked destinations</a> <a for=list>contains</a>
+     |request|'s <a for=request>destination</a>,
+     set |block| to true.
+  1. If |policy|'s <a>sources</a> <a for=list>contains</a> "`inline`"
+     and |report policy|'s <a>blocked destinations</a> <a for=list>contains</a>
+     |request|'s <a for=request>destination</a>,
+     set |report block| to true.
+  1. If |block| or |report block|, <a>report violation</a> with |request|, |block|, |report block|,
```

Comment:
Usually we write out the if conditions for clarity, including for booleans.

Reply:
Done!